contracts/lists/src/registrations.rs

```diff
@@ -0,0 +1,544 @@
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, PartialEq, Debug, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct RegistrationInternal {
+    // don't need to store ID since it's the key
+    pub registrant_id: AccountId,
+    pub list_id: ListId,
+    pub status: RegistrationStatus,
+    pub submitted_ms: TimestampMs,
+    pub updated_ms: TimestampMs,
+    pub admin_notes: Option<String>,
+    pub registrant_notes: Option<String>,
+    pub registered_by: Option<AccountId>, // Could be list owner or a list admin. If None, use registrant_id. Used for processing refund on deletion of registration.
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub enum VersionedRegistrationInternal {
+    Current(RegistrationInternal),
+}
+
+impl From<VersionedRegistrationInternal> for RegistrationInternal {
+    fn from(registration_internal: VersionedRegistrationInternal) -> Self {
+        match registration_internal {
+            VersionedRegistrationInternal::Current(current) => current,
+        }
+    }
+}
+
+// Ephemeral data structure used for view methods, not stored within contract
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, PartialEq, Debug, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct RegistrationExternal {
+    pub id: RegistrationId,
+    pub registrant_id: AccountId,
+    pub list_id: ListId,
+    pub status: RegistrationStatus,
+    pub submitted_ms: TimestampMs,
+    pub updated_ms: TimestampMs,
+    pub admin_notes: Option<String>,
+    pub registrant_notes: Option<String>,
+    pub registered_by: AccountId,
+}
+
+pub(crate) fn format_registration(
+    registration_id: RegistrationId,
+    registration_internal: RegistrationInternal,
+) -> RegistrationExternal {
+    RegistrationExternal {
+        id: registration_id,
+        registrant_id: registration_internal.registrant_id.clone(),
+        list_id: registration_internal.list_id,
+        status: registration_internal.status,
+        submitted_ms: registration_internal.submitted_ms,
+        updated_ms: registration_internal.updated_ms,
+        admin_notes: registration_internal.admin_notes,
+        registrant_notes: registration_internal.registrant_notes,
+        registered_by: registration_internal
+            .registered_by
+            .unwrap_or_else(|| registration_internal.registrant_id.clone()),
+    }
+}
+
+// Ephemeral struct for admins to use when registering a registrant
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, PartialEq, Debug, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct RegistrationInput {
+    pub registrant_id: AccountId,
+    pub status: RegistrationStatus,
+    pub submitted_ms: Option<TimestampMs>,
+    pub updated_ms: Option<TimestampMs>,
+    pub notes: Option<String>,
+}
+
+#[near_bindgen]
+impl Contract {
+    pub(crate) fn register(
+        &mut self,
+        list_id: ListId,
+        list: ListInternal,
```

Comment:
so i understand that this function is called by the `register_batch` function, but it also allows for a standalone call, i think it might introduce a scenario(through the standalone call) where a user can pass approved as the default status of the `ListInternal` and hence could get their registration approved straight from registration(like if an admin did it).
please check and see if my assumption is correct.

Reply:
It can't be called publicly because it's only public to the crate (so it can only be called from within the contract)

Reply:
ah, missed that. 👍